internal/jobmanager/jobmanager_test.go

```diff
@@ -0,0 +1,1031 @@
+package jobmanager
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/edulinq/autograder/internal/util"
+)
+
+var testLockKey string = "test_key"
+var testPoolSize int = 1
+
+var input []string = []string{
+	"A",
+	"BB",
+	"CCC",
+}
+
+var workFunc = func(input string) (int, error) {
+	return len(input), nil
+}
+
+type printableJob[InputType any, OutputType any] struct {
```

Comment:
Can you just get away with this by putting the JSON tags on a normal job (and omitting `"-"`) the funcs?